

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Example from mouse to human &mdash; TransBrain 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../_static/nbsphinx-code-cells.css?v=2aa19091" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js?v=e031e9a9"></script>
      <script src="../_static/doctools.js?v=888ff710"></script>
      <script src="../_static/sphinx_highlight.js?v=4825356b"></script>
      <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
      <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            TransBrain
              <img src="../_static/logo_final.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Setup &amp; Usage</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../installation.html#installation-options">Installation Options</a></li>
<li class="toctree-l2"><a class="reference internal" href="../installation.html#python-dependencies">Python Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../installation.html#unit-test">Unit Test</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../user.html">User Guide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../user.html#id2">General Use</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user.html#id3">Online Mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user.html#id4">Advanced Use</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../building.html">Construction Pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../pipeline/run_pipeline/integration.html">Transcriptional Data Integration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Selection-of-Cross-Modality-Stable-Genes">Selection of Cross-Modality Stable Genes</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#The-shared-gene-space-between-the-​​AHBA,-sn-dataset-and-​​mouse-st-dataset​​">The shared gene space between the ​​AHBA, sn dataset and ​​mouse st dataset​​</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Cortical-stable-genes">Cortical stable genes</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Subcortical-stable-genes">Subcortical stable genes</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#All-stable-genes">All stable genes</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Determining-Similarity-Thresholds-for-Sample-Pool-Construction">Determining Similarity Thresholds for Sample Pool Construction</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Thresholds-of-cortex">Thresholds of cortex</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Thresholds-of-subcortex">Thresholds of subcortex</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Thresholds-of-all">Thresholds of all</a></li>
<li class="toctree-l4"><a class="reference internal" href="../pipeline/run_pipeline/integration.html#Integrating-dataset">Integrating dataset</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../pipeline/run_pipeline/dnn.html">DNN Model Training</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/dnn.html#cortical-train">cortical train</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/dnn.html#subcortical-train">subcortical train</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/dnn.html#all-train">all train</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../pipeline/run_pipeline/graph_embedding.html">Graph-based Random Walk</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/graph_embedding.html#Define-human-and-mouse-ROIs">Define human and mouse ROIs</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/graph_embedding.html#Generate-human-and-mouse-graph">Generate human and mouse graph</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/graph_embedding.html#Hierachy-constraint">Hierachy constraint</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pipeline/run_pipeline/graph_embedding.html#Generate-graph-embeddings">Generate graph embeddings</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../methods/mapping.html">Bidirectional Mapping</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../cases.html">Example Cases</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html">Analyzing cross-species similarity of transcriptional embeddings</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Human-Autocorrelation-Matrix">Human Autocorrelation Matrix</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Cross-species-Similarity-Matrix">Cross-species Similarity Matrix</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Potential-homologous-regions'-rank-of-Cross-species-Matrix">Potential homologous regions’ rank of Cross-species Matrix</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Cortical-rank">Cortical rank</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Subcortical-rank">Subcortical rank</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Gradients-of-TRembeddings-conserved-across-species">Gradients of TRembeddings conserved across species</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Load-cortical-embeddings">Load cortical embeddings</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#PCA-of-Cortical-embeddings">PCA of Cortical embeddings</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Visulization">Visulization</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient1">Human Gradient1</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient2">Human Gradient2</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient3">Human Gradient3</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient1">Mouse gradient1</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient2">Mouse gradient2</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient3">Mouse gradient3</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html">Inferring the conservation of resting-fMRI gradients across species</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#translating-mouse-gradient-to-human">translating mouse gradient to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#visulization">visulization</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#mouse-gradient">mouse gradient</a></li>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#mouse-gradients-mapped-to-human">mouse gradients mapped to human</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#human-gradient">human gradient</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/gradient_spectrum/fMRI_gradients.html#correlation">correlation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../tutorials/notebooks/translation/optogenetic_annotation/optogenetic_circuits.html">Annotating optogenetic circuits</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/optogenetic_annotation/optogenetic_circuits.html#mouse-opto-phenotypes">mouse opto phenotypes</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/optogenetic_annotation/optogenetic_circuits.html#translating-mouse-ai-stimulated-pattern-to-human">translating mouse ai stimulated pattern to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/optogenetic_annotation/optogenetic_circuits.html#translating-mouse-drn-stimulated-pattern-to-human">translating mouse drn stimulated pattern to human</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../tutorials/notebooks/translation/optogenetic_annotation/optogenetic_circuits.html#annotating-using-NeuroSynth.">annotating using NeuroSynth.</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html">Linking gene mutations to imaging phenotype deviations in autism</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#load-deviation-pattern">load deviation pattern</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#translating-mouse-mutation-patterns-to-human">translating mouse mutation patterns to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#individual-gene-risk-scores-based-on-mouse-model">individual gene risk scores based on mouse model</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#load-AHBA-gene-expression">load AHBA gene expression</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#individual-gene-risk-scores-based-on-AHBA">individual gene risk scores based on AHBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#correlate-risk-scores">correlate risk scores</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#correlate-risk-scores-in-specific-gene">correlate risk scores in specific gene</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#correlating-sythetic-mutation-pattern-with-AHBA-gene-expression">correlating sythetic mutation pattern with AHBA gene expression</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/notebooks/translation/autism_mutation/gene_mutations.html#load-behavior-data">load behavior data</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../api/utils.html">Utils</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../api/data_process_api.html">Transcriptional Data Process</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_qc"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_qc()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_normalized"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_normalized()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_common_genes"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_common_genes()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_batch_correct"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_batch_correct()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_smooth"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_smooth()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_scaled"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_scaled()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.preprocess_pipeline"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.preprocess_pipeline()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../api/data_integration_api.html">Data Integration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.utils.stable_gene_filter"><code class="docutils literal notranslate"><span class="pre">stable_gene_filter()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.utils.calculate_thresholds"><code class="docutils literal notranslate"><span class="pre">calculate_thresholds()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator"><code class="docutils literal notranslate"><span class="pre">DataIntegrator</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.__init__"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.__init__()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.run"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.run()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.pool_construction"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.pool_construction()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.clean_pool"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.clean_pool()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.weighted_average"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.weighted_average()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.downsample_data"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.downsample_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.generate_integrated_anndata"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.generate_integrated_anndata()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../api/dnn_api.html">DNN</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.analyzer.train_test_split"><code class="docutils literal notranslate"><span class="pre">train_test_split()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.analyzer.calculate_rank"><code class="docutils literal notranslate"><span class="pre">calculate_rank()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.analyzer.transform_human_mouse"><code class="docutils literal notranslate"><span class="pre">transform_human_mouse()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.analyzer.independent_test"><code class="docutils literal notranslate"><span class="pre">independent_test()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline"><code class="docutils literal notranslate"><span class="pre">DnnPipeline</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.cfg"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.cfg</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.logger"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.logger</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.execute"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.execute()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.train"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.train()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.trans"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.trans()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.test"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline._process_single_dataset"><code class="docutils literal notranslate"><span class="pre">DnnPipeline._process_single_dataset()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline._train_single_model"><code class="docutils literal notranslate"><span class="pre">DnnPipeline._train_single_model()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../api/graph_walk_api.html">Graph Walk</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/graph_walk_api.html#pipeline.code.graph_walk.run.generate_embeddings"><code class="docutils literal notranslate"><span class="pre">generate_embeddings()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../api/external_api.html">API Reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../api/fetch_api.html">Fetching Atlases</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/fetch_api.html#transbrain.atlas.fetch_human_atlas"><code class="docutils literal notranslate"><span class="pre">fetch_human_atlas()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../api/fetch_api.html#transbrain.atlas.fetch_mouse_atlas"><code class="docutils literal notranslate"><span class="pre">fetch_mouse_atlas()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../api/base_api.html">Base Functions</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/base_api.html#transbrain.base.get_region_phenotypes"><code class="docutils literal notranslate"><span class="pre">get_region_phenotypes()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../api/mapping_api.html">Mapping Functions</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans.atlas_type"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.atlas_type</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans.regions"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.regions</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans.embeddings"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.embeddings</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans._load_embeddings"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._load_embeddings()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans._dual_mapping"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._dual_mapping()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans._translate"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._translate()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans.mouse_to_human"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.mouse_to_human()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../api/mapping_api.html#transbrain.trans.SpeciesTrans.human_to_mouse"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.human_to_mouse()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Other</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../citing.html">Citing TransBrain</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">TransBrain</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Example from mouse to human</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/tests/basic_usage.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">transbrain</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">tb</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">os</span>
</pre></div>
</div>
</div>
<section id="Example-from-mouse-to-human">
<h1>Example from mouse to human<a class="headerlink" href="#Example-from-mouse-to-human" title="Permalink to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Initialize TransBrain for bn atlas</span>
<span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>

<span class="n">mouse_phenotype</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../transbrain/exampledata/mouse/mouse_all_example_data.csv&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">mouse_phenotype_in_human</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">mouse_to_human</span><span class="p">(</span><span class="n">mouse_phenotype</span><span class="p">,</span> <span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">normalize</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">mouse_phenotype_in_human</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
INFO:root:Successfully translated mouse all phenotypes to human.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
        Phenotype
A8m     -0.036860
A8dl    -0.037950
A9l     -0.048863
A6dl    -0.023963
A6m     -0.005832
...           ...
Otha     0.286257
mPFtha   0.568260
lPFtha   0.224642
rTtha    0.732512
PPtha    0.469093

[127 rows x 1 columns]
</pre></div></div>
</div>
</section>
<section id="Example-from-human-to-mouse">
<h1>Example from human to mouse<a class="headerlink" href="#Example-from-human-to-mouse" title="Permalink to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>

<span class="n">human_phenotype</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../transbrain/exampledata/human/bn/human_bn_all_example_data.csv&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">human_phenotype_in_mouse</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">human_to_mouse</span><span class="p">(</span><span class="n">human_phenotype</span><span class="p">,</span> <span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">normalize</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">human_phenotype_in_mouse</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
INFO:root:Successfully translated human all phenotypes to mouse.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
      Phenotype
ACAd   0.296680
ACAv   0.303368
PL     0.327434
ILA    0.332395
ORBl   0.386484
...         ...
MTN    0.696938
ILM    0.688610
GENv   0.638566
EPI    0.731161
RT     0.640969

[66 rows x 1 columns]
</pre></div></div>
</div>
</section>
<section id="Get-phenotypes-in-Human-atlas-used-in-TransBrain">
<h1>Get phenotypes in Human atlas used in TransBrain<a class="headerlink" href="#Get-phenotypes-in-Human-atlas-used-in-TransBrain" title="Permalink to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_atlas</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">atlas</span><span class="o">.</span><span class="n">fetch_human_atlas</span><span class="p">(</span><span class="n">atlas_type</span><span class="o">=</span><span class="s1">&#39;bn&#39;</span><span class="p">,</span><span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;cortex&#39;</span><span class="p">)</span>
<span class="n">phenotype_nii_path</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;../transbrain/exampledata/human/human_example_phenotype_data.nii.gz&#39;</span><span class="p">)</span>
<span class="n">human_phenptype_extracted</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">base</span><span class="o">.</span><span class="n">get_region_phenotypes</span><span class="p">(</span><span class="n">phenotype_nii_path</span><span class="p">,</span> <span class="n">atlas_dict</span> <span class="o">=</span> <span class="n">human_atlas</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">human_phenptype_extracted</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
             Phenotype
Region name
A8m           0.064848
A8dl          1.543472
A9l           1.847849
A6dl         -0.452153
A6m          -1.175266
...                ...
A28/34        1.185400
TI           -0.214262
TH           -0.619775
rpSTS         0.218751
cpSTS        -0.438712

[105 rows x 1 columns]
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Mapping</span>
<span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>
<span class="n">human_phenotype_in_mouse</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">human_to_mouse</span><span class="p">(</span><span class="n">human_phenptype_extracted</span><span class="p">,</span> <span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;cortex&#39;</span><span class="p">,</span> <span class="n">normalize</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">human_phenotype_in_mouse</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
INFO:root:Successfully translated human cortex phenotypes to mouse.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
         Phenotype
ACAd      0.752345
ACAv      0.808829
PL        0.656465
ILA       0.514904
ORBl      0.805533
ORBm      0.661180
ORBvl     0.873377
MOp       0.017489
SSp-n     0.137524
SSp-bfd   0.163902
SSp-ll    0.097725
SSp-m    -0.027497
SSp-ul   -0.029382
SSp-tr    0.081100
SSp-un    0.297645
SSs       0.139606
PTLp      0.058395
RSPagl    0.223968
RSPd      0.381135
RSPv      0.279498
VISpm     0.140252
VISp      0.051762
VISal     0.030422
VISl      0.077017
VISpl     0.073758
AId       0.054786
AIp       0.009841
AIv       0.068754
GU        0.008394
VISC     -0.025568
TEa       0.469960
PERI      0.514153
ECT       0.404478
AUDd      0.336839
AUDp      0.550344
AUDpo     0.486819
AUDv      0.562850
</pre></div></div>
</div>
</section>
<section id="Get-phenotypes-in-Mouse-atlas-used-in-TransBrain">
<h1>Get phenotypes in Mouse atlas used in TransBrain<a class="headerlink" href="#Get-phenotypes-in-Mouse-atlas-used-in-TransBrain" title="Permalink to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mouse_atlas</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">atlas</span><span class="o">.</span><span class="n">fetch_mouse_atlas</span><span class="p">(</span><span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">)</span>
<span class="n">phenotype_nii_path</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;../transbrain/exampledata/mouse/mouse_example_phenotype_data.nii.gz&#39;</span><span class="p">)</span>
<span class="n">mouse_phenptype_extracted</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">base</span><span class="o">.</span><span class="n">get_region_phenotypes</span><span class="p">(</span><span class="n">phenotype_nii_path</span><span class="p">,</span> <span class="n">atlas_dict</span> <span class="o">=</span> <span class="n">mouse_atlas</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">mouse_phenptype_extracted</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
             Phenotype
Region name
ACAd          0.604520
ACAv          0.369486
AId           4.316058
AIp           2.043769
AIv           4.201185
...                ...
MTN           4.848961
ILM           5.223915
GENv          0.211644
EPI           3.944127
RT            1.150919

[68 rows x 1 columns]
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Mapping</span>
<span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>
<span class="n">mouse_phenotype_in_human</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">mouse_to_human</span><span class="p">(</span><span class="n">mouse_phenptype_extracted</span><span class="p">,</span> <span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">normalize</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">mouse_phenotype_in_human</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
INFO:root:Successfully translated mouse all phenotypes to human.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
        Phenotype
A8m     -0.036852
A8dl    -0.037943
A9l     -0.048855
A6dl    -0.023955
A6m     -0.005824
...           ...
Otha     0.286260
mPFtha   0.568262
lPFtha   0.224644
rTtha    0.732510
PPtha    0.469096

[127 rows x 1 columns]
</pre></div></div>
</div>
</section>
<section id="Get-graphembeddings">
<h1>Get graphembeddings<a class="headerlink" href="#Get-graphembeddings" title="Permalink to this heading"></a></h1>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>
<span class="n">Human_Mouse_embedding_bn</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">_load_embeddings</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">Human_Mouse_embedding_bn</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
(500, 193, 40)
</pre></div></div>
</div>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, TransBrain Developers.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>