

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Linking gene mutations to imaging phenotype deviations in autism &mdash; TransBrain 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/nbsphinx-code-cells.css?v=2aa19091" />

  
      <script src="../../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js?v=e031e9a9"></script>
      <script src="../../../../_static/doctools.js?v=888ff710"></script>
      <script src="../../../../_static/sphinx_highlight.js?v=4825356b"></script>
      <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
      <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="Utils" href="../../../../api/utils.html" />
    <link rel="prev" title="Annotating optogenetic circuits" href="../optogenetic_annotation/optogenetic_circuits.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../../index.html" class="icon icon-home">
            TransBrain
              <img src="../../../../_static/logo_final.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Setup &amp; Usage</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../installation.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../installation.html#installation-options">Installation Options</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../installation.html#python-dependencies">Python Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../installation.html#unit-test">Unit Test</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../user.html">User Guide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../user.html#id2">General Use</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../user.html#id3">Online Mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../user.html#id4">Advanced Use</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../building.html">Construction Pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html">Transcriptional Data Integration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Selection-of-Cross-Modality-Stable-Genes">Selection of Cross-Modality Stable Genes</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#The-shared-gene-space-between-the-​​AHBA,-sn-dataset-and-​​mouse-st-dataset​​">The shared gene space between the ​​AHBA, sn dataset and ​​mouse st dataset​​</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Cortical-stable-genes">Cortical stable genes</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Subcortical-stable-genes">Subcortical stable genes</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#All-stable-genes">All stable genes</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Determining-Similarity-Thresholds-for-Sample-Pool-Construction">Determining Similarity Thresholds for Sample Pool Construction</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Thresholds-of-cortex">Thresholds of cortex</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Thresholds-of-subcortex">Thresholds of subcortex</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Thresholds-of-all">Thresholds of all</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../pipeline/run_pipeline/integration.html#Integrating-dataset">Integrating dataset</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../pipeline/run_pipeline/dnn.html">DNN Model Training</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/dnn.html#cortical-train">cortical train</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/dnn.html#subcortical-train">subcortical train</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/dnn.html#all-train">all train</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../pipeline/run_pipeline/graph_embedding.html">Graph-based Random Walk</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/graph_embedding.html#Define-human-and-mouse-ROIs">Define human and mouse ROIs</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/graph_embedding.html#Generate-human-and-mouse-graph">Generate human and mouse graph</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/graph_embedding.html#Hierachy-constraint">Hierachy constraint</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../pipeline/run_pipeline/graph_embedding.html#Generate-graph-embeddings">Generate graph embeddings</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../methods/mapping.html">Bidirectional Mapping</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../../cases.html">Example Cases</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html">Analyzing cross-species similarity of transcriptional embeddings</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Human-Autocorrelation-Matrix">Human Autocorrelation Matrix</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Cross-species-Similarity-Matrix">Cross-species Similarity Matrix</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Potential-homologous-regions'-rank-of-Cross-species-Matrix">Potential homologous regions’ rank of Cross-species Matrix</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Cortical-rank">Cortical rank</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Subcortical-rank">Subcortical rank</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Gradients-of-TRembeddings-conserved-across-species">Gradients of TRembeddings conserved across species</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Load-cortical-embeddings">Load cortical embeddings</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#PCA-of-Cortical-embeddings">PCA of Cortical embeddings</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Visulization">Visulization</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient1">Human Gradient1</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient2">Human Gradient2</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Human-Gradient3">Human Gradient3</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient1">Mouse gradient1</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient2">Mouse gradient2</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../tr_embeddings_analysis/embeddings_analysis.html#Mouse-gradient3">Mouse gradient3</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html">Inferring the conservation of resting-fMRI gradients across species</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#translating-mouse-gradient-to-human">translating mouse gradient to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#visulization">visulization</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#mouse-gradient">mouse gradient</a></li>
<li class="toctree-l4"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#mouse-gradients-mapped-to-human">mouse gradients mapped to human</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#human-gradient">human gradient</a></li>
<li class="toctree-l3"><a class="reference internal" href="../gradient_spectrum/fMRI_gradients.html#correlation">correlation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../optogenetic_annotation/optogenetic_circuits.html">Annotating optogenetic circuits</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../optogenetic_annotation/optogenetic_circuits.html#mouse-opto-phenotypes">mouse opto phenotypes</a></li>
<li class="toctree-l3"><a class="reference internal" href="../optogenetic_annotation/optogenetic_circuits.html#translating-mouse-ai-stimulated-pattern-to-human">translating mouse ai stimulated pattern to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="../optogenetic_annotation/optogenetic_circuits.html#translating-mouse-drn-stimulated-pattern-to-human">translating mouse drn stimulated pattern to human</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../optogenetic_annotation/optogenetic_circuits.html#annotating-using-NeuroSynth.">annotating using NeuroSynth.</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Linking gene mutations to imaging phenotype deviations in autism</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#load-deviation-pattern">load deviation pattern</a></li>
<li class="toctree-l3"><a class="reference internal" href="#translating-mouse-mutation-patterns-to-human">translating mouse mutation patterns to human</a></li>
<li class="toctree-l3"><a class="reference internal" href="#individual-gene-risk-scores-based-on-mouse-model">individual gene risk scores based on mouse model</a></li>
<li class="toctree-l3"><a class="reference internal" href="#load-AHBA-gene-expression">load AHBA gene expression</a></li>
<li class="toctree-l3"><a class="reference internal" href="#individual-gene-risk-scores-based-on-AHBA">individual gene risk scores based on AHBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="#correlate-risk-scores">correlate risk scores</a></li>
<li class="toctree-l3"><a class="reference internal" href="#correlate-risk-scores-in-specific-gene">correlate risk scores in specific gene</a></li>
<li class="toctree-l3"><a class="reference internal" href="#correlating-sythetic-mutation-pattern-with-AHBA-gene-expression">correlating sythetic mutation pattern with AHBA gene expression</a></li>
<li class="toctree-l3"><a class="reference internal" href="#load-behavior-data">load behavior data</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/utils.html">Utils</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/data_process_api.html">Transcriptional Data Process</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_qc"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_qc()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_normalized"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_normalized()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_common_genes"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_common_genes()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_batch_correct"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_batch_correct()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_smooth"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_smooth()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.dataset_scaled"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.dataset_scaled()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_process_api.html#pipeline.code.dpreprocess.sn_.Preprocess_Sn.preprocess_pipeline"><code class="docutils literal notranslate"><span class="pre">Preprocess_Sn.preprocess_pipeline()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/data_integration_api.html">Data Integration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.utils.stable_gene_filter"><code class="docutils literal notranslate"><span class="pre">stable_gene_filter()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.utils.calculate_thresholds"><code class="docutils literal notranslate"><span class="pre">calculate_thresholds()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator"><code class="docutils literal notranslate"><span class="pre">DataIntegrator</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.__init__"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.__init__()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.run"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.run()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.pool_construction"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.pool_construction()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.clean_pool"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.clean_pool()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.weighted_average"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.weighted_average()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.downsample_data"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.downsample_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/data_integration_api.html#pipeline.code.dintegrate.main.DataIntegrator.generate_integrated_anndata"><code class="docutils literal notranslate"><span class="pre">DataIntegrator.generate_integrated_anndata()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/dnn_api.html">DNN</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.analyzer.train_test_split"><code class="docutils literal notranslate"><span class="pre">train_test_split()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.analyzer.calculate_rank"><code class="docutils literal notranslate"><span class="pre">calculate_rank()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.analyzer.transform_human_mouse"><code class="docutils literal notranslate"><span class="pre">transform_human_mouse()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.analyzer.independent_test"><code class="docutils literal notranslate"><span class="pre">independent_test()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline"><code class="docutils literal notranslate"><span class="pre">DnnPipeline</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.cfg"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.cfg</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.logger"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.logger</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.execute"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.execute()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.train"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.train()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.trans"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.trans()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline.test"><code class="docutils literal notranslate"><span class="pre">DnnPipeline.test()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline._process_single_dataset"><code class="docutils literal notranslate"><span class="pre">DnnPipeline._process_single_dataset()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/dnn_api.html#pipeline.code.dnn.pipeline.DnnPipeline._train_single_model"><code class="docutils literal notranslate"><span class="pre">DnnPipeline._train_single_model()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/graph_walk_api.html">Graph Walk</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/graph_walk_api.html#pipeline.code.graph_walk.run.generate_embeddings"><code class="docutils literal notranslate"><span class="pre">generate_embeddings()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/external_api.html">API Reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/fetch_api.html">Fetching Atlases</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/fetch_api.html#transbrain.atlas.fetch_human_atlas"><code class="docutils literal notranslate"><span class="pre">fetch_human_atlas()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/fetch_api.html#transbrain.atlas.fetch_mouse_atlas"><code class="docutils literal notranslate"><span class="pre">fetch_mouse_atlas()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/base_api.html">Base Functions</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/base_api.html#transbrain.base.get_region_phenotypes"><code class="docutils literal notranslate"><span class="pre">get_region_phenotypes()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../api/mapping_api.html">Mapping Functions</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans.atlas_type"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.atlas_type</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans.regions"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.regions</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans.embeddings"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.embeddings</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans._load_embeddings"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._load_embeddings()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans._dual_mapping"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._dual_mapping()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans._translate"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans._translate()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans.mouse_to_human"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.mouse_to_human()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../../../../api/mapping_api.html#transbrain.trans.SpeciesTrans.human_to_mouse"><code class="docutils literal notranslate"><span class="pre">SpeciesTrans.human_to_mouse()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Other</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citing.html">Citing TransBrain</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">TransBrain</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../../cases.html">Example Cases</a></li>
      <li class="breadcrumb-item active">Linking gene mutations to imaging phenotype deviations in autism</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../_sources/tutorials/notebooks/translation/autism_mutation/gene_mutations.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="Linking-gene-mutations-to-imaging-phenotype-deviations-in-autism">
<h1>Linking gene mutations to imaging phenotype deviations in autism<a class="headerlink" href="#Linking-gene-mutations-to-imaging-phenotype-deviations-in-autism" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p>This example shows how to translate mouse mutation patterns to human and compare with imaging phenotype deviations in autism.</p></li>
<li><p>See the this <a class="reference internal" href="case4_intro.html"><span class="doc">page</span></a> for more introduction of this tutorial.</p></li>
</ul>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w">  </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">nilearn</span><span class="w"> </span><span class="kn">import</span> <span class="n">image</span><span class="p">,</span><span class="n">plotting</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">scipy</span><span class="w"> </span><span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">seaborn</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sns</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pickle</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">os</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">transbrain</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">tb</span>
</pre></div>
</div>
</div>
<section id="load-deviation-pattern">
<h2>load deviation pattern<a class="headerlink" href="#load-deviation-pattern" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_deviation_pattern</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;z_autism_regress.csv&#39;</span><span class="p">)</span> <span class="c1"># Deviant patterns in autistic individuals</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mouse_mutation_pattern</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;mouse_mutation_pattern.csv&#39;</span><span class="p">)</span> <span class="c1"># Deviant patterns in autism mouse models</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mouse_mutation_pattern</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</section>
<section id="translating-mouse-mutation-patterns-to-human">
<h2>translating mouse mutation patterns to human<a class="headerlink" href="#translating-mouse-mutation-patterns-to-human" title="Permalink to this heading"></a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">Transformer</span> <span class="o">=</span> <span class="n">tb</span><span class="o">.</span><span class="n">trans</span><span class="o">.</span><span class="n">SpeciesTrans</span><span class="p">(</span><span class="s1">&#39;bn&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Initialized for bn atlas.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">Transformer</span><span class="o">.</span><span class="n">mouse_to_human</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on method mouse_to_human in module transbrain.trans:

mouse_to_human(phenotype: pandas.core.frame.DataFrame, region_type: Literal[&#39;cortex&#39;, &#39;subcortex&#39;, &#39;all&#39;] = &#39;cortex&#39;, normalize: bool = True, restore: bool = False) -&gt; pandas.core.frame.DataFrame method of transbrain.trans.SpeciesTrans instance
    Translate mouse phenotype to human.

    Parameters
    ----------
    phenotype : pd.DataFrame
        Mouse phenotype DataFrame (regions × phenotypes).
    region_type : {&#39;cortex&#39;, &#39;subcortex&#39;, &#39;all&#39;}, optional
        The brain region type to translate. Default is &#39;cortex&#39;.
    normalize : bool, optional
        Whether to normalize data before translation. Default is True.
    restore : bool, optional
        Whether to restore values back to original scale after translation. Only used if normalize is True.
        Please enable this parameter with caution, unless you are certain that the distributions of this phenotype are consistent between the two species.

    Returns
    -------
    pd.DataFrame
        Translated human phenotype DataFrame (regions × phenotypes).

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mutation_trans_in_human</span> <span class="o">=</span> <span class="n">Transformer</span><span class="o">.</span><span class="n">mouse_to_human</span><span class="p">(</span><span class="n">mouse_mutation_pattern</span><span class="p">,</span><span class="n">region_type</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:root:Successfully translated mouse all phenotypes to human.
</pre></div></div>
</div>
</section>
<section id="individual-gene-risk-scores-based-on-mouse-model">
<h2>individual gene risk scores based on mouse model<a class="headerlink" href="#individual-gene-risk-scores-based-on-mouse-model" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cross_species_autism_dataframe</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]):</span>

    <span class="n">row_data</span> <span class="o">=</span> <span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">corr_l</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">5</span><span class="p">):</span>

        <span class="n">m_data</span> <span class="o">=</span> <span class="n">mutation_trans_in_human</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
        <span class="n">corr_</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">row_data</span><span class="p">,</span><span class="n">m_data</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">corr_l</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">corr_</span><span class="p">)</span>

    <span class="n">cross_species_autism_dataframe</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="n">corr_l</span>

<span class="n">cross_species_autism_dataframe</span> <span class="o">=</span> <span class="n">cross_species_autism_dataframe</span><span class="o">.</span><span class="n">T</span>
<span class="n">cross_species_autism_dataframe</span><span class="o">.</span><span class="n">index</span> <span class="o">=</span> <span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">index</span>
<span class="n">cross_species_autism_dataframe</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">mutation_trans_in_human</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</section>
<section id="load-AHBA-gene-expression">
<h2>load AHBA gene expression<a class="headerlink" href="#load-AHBA-gene-expression" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">AHBA_ex</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;ahba_expression.csv&#39;</span><span class="p">)</span> <span class="c1"># AHBA expression in region level</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">AHBA_ex</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</section>
<section id="individual-gene-risk-scores-based-on-AHBA">
<h2>individual gene risk scores based on AHBA<a class="headerlink" href="#individual-gene-risk-scores-based-on-AHBA" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">AHBA_autism_dataframe</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]):</span>

    <span class="n">row_data</span> <span class="o">=</span> <span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">corr_l</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">5</span><span class="p">):</span>

        <span class="n">m_data</span> <span class="o">=</span> <span class="n">AHBA_ex</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
        <span class="n">corr_</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">row_data</span><span class="p">,</span><span class="n">m_data</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">corr_l</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">corr_</span><span class="p">)</span>

    <span class="n">AHBA_autism_dataframe</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="n">corr_l</span>

<span class="n">AHBA_autism_dataframe</span> <span class="o">=</span> <span class="n">AHBA_autism_dataframe</span><span class="o">.</span><span class="n">T</span>
<span class="n">AHBA_autism_dataframe</span><span class="o">.</span><span class="n">index</span> <span class="o">=</span> <span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">index</span>
<span class="n">AHBA_autism_dataframe</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">mutation_trans_in_human</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
</section>
<section id="correlate-risk-scores">
<h2>correlate risk scores<a class="headerlink" href="#correlate-risk-scores" title="Permalink to this heading"></a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">AHBA_autism_dataframe</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span><span class="n">cross_species_autism_dataframe</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PearsonRResult(statistic=0.19780599996495293, pvalue=9.645342333400538e-12)
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span><span class="mi">3</span><span class="p">),</span><span class="n">dpi</span><span class="o">=</span><span class="mi">400</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">labelsize</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>

<span class="k">for</span> <span class="n">spine</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">spine</span><span class="o">.</span><span class="n">set_position</span><span class="p">((</span><span class="s1">&#39;outward&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mf">0.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="mf">0.3</span><span class="p">])</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="o">-</span><span class="mf">0.4</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="mf">0.4</span><span class="p">])</span>

<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">25</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">regplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">AHBA_autism_dataframe</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span><span class="n">y</span><span class="o">=</span><span class="n">cross_species_autism_dataframe</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="n">scatter_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;color&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;linewidths&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span><span class="s1">&#39;alpha&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="s1">&#39;s&#39;</span><span class="p">:</span><span class="mi">40</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_21_0.png" src="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_21_0.png" />
</div>
</div>
</section>
<section id="correlate-risk-scores-in-specific-gene">
<h2>correlate risk scores in specific gene<a class="headerlink" href="#correlate-risk-scores-in-specific-gene" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;Dvl1&#39;</span><span class="p">,</span><span class="s1">&#39;Magel2&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a4&#39;</span><span class="p">,</span><span class="s1">&#39;Sgsh&#39;</span><span class="p">,</span><span class="s1">&#39;Shank3--&#39;</span><span class="p">]:</span>

    <span class="n">AHBA_scores</span> <span class="o">=</span> <span class="n">AHBA_autism_dataframe</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">mouse_mutation_scores</span> <span class="o">=</span> <span class="n">cross_species_autism_dataframe</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">AHBA_scores</span><span class="p">,</span><span class="n">mouse_mutation_scores</span><span class="p">)</span>

    <span class="n">dict_gene_</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{&#39;Dvl1&#39;: (0.31097966299833013, 1.2856342421455238e-06),
 &#39;Magel2&#39;: (0.4539929971213872, 3.0064151593370794e-13),
 &#39;Slc6a4&#39;: (0.015197117448081244, 0.8175170464931003),
 &#39;Sgsh&#39;: (0.002248242177818731, 0.9727707705317751),
 &#39;Shank3--&#39;: (0.20016237908524429, 0.0021402800433741828)}
</pre></div></div>
</div>
</section>
<section id="correlating-sythetic-mutation-pattern-with-AHBA-gene-expression">
<h2>correlating sythetic mutation pattern with AHBA gene expression<a class="headerlink" href="#correlating-sythetic-mutation-pattern-with-AHBA-gene-expression" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">gene1</span><span class="p">,</span><span class="n">gene2</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">([</span><span class="s1">&#39;DVL1&#39;</span><span class="p">,</span><span class="s1">&#39;MAGEL2&#39;</span><span class="p">,</span><span class="s1">&#39;SLC6A4&#39;</span><span class="p">,</span><span class="s1">&#39;SGSH&#39;</span><span class="p">,</span><span class="s1">&#39;SHANK3&#39;</span><span class="p">],[</span><span class="s1">&#39;Dvl1&#39;</span><span class="p">,</span><span class="s1">&#39;Magel2&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a4&#39;</span><span class="p">,</span><span class="s1">&#39;Sgsh&#39;</span><span class="p">,</span><span class="s1">&#39;Shank3--&#39;</span><span class="p">]):</span>
    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">AHBA_ex</span><span class="o">.</span><span class="n">T</span><span class="p">[</span><span class="n">gene1</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">,</span><span class="n">mutation_trans_in_human</span><span class="p">[</span><span class="n">gene2</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">)</span>
    <span class="n">dict_gene_</span><span class="p">[</span><span class="n">gene1</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{&#39;DVL1&#39;: (0.09730098359735115, 0.2764762751949792),
 &#39;MAGEL2&#39;: (0.3526186809312027, 4.78449085894465e-05),
 &#39;SLC6A4&#39;: (-0.062062969713091054, 0.4882044955368892),
 &#39;SGSH&#39;: (0.04501704457506566, 0.6152770360499245),
 &#39;SHANK3&#39;: (-0.06759348765582845, 0.4502055153591825)}
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">gene1</span><span class="p">,</span><span class="n">gene2</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">([</span><span class="s1">&#39;DVL1&#39;</span><span class="p">,</span><span class="s1">&#39;MAGEL2&#39;</span><span class="p">,</span><span class="s1">&#39;SLC6A4&#39;</span><span class="p">,</span><span class="s1">&#39;SGSH&#39;</span><span class="p">,</span><span class="s1">&#39;SHANK3&#39;</span><span class="p">],[</span><span class="s1">&#39;Dvl1&#39;</span><span class="p">,</span><span class="s1">&#39;Magel2&#39;</span><span class="p">,</span><span class="s1">&#39;Slc6a4&#39;</span><span class="p">,</span><span class="s1">&#39;Sgsh&#39;</span><span class="p">,</span><span class="s1">&#39;Shank3--&#39;</span><span class="p">]):</span>
    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">AHBA_ex</span><span class="o">.</span><span class="n">T</span><span class="p">[</span><span class="n">gene1</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">[:</span><span class="mi">105</span><span class="p">],</span><span class="n">mutation_trans_in_human</span><span class="p">[</span><span class="n">gene2</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">[:</span><span class="mi">105</span><span class="p">])</span>
    <span class="n">dict_gene_</span><span class="p">[</span><span class="n">gene1</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dict_gene_</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{&#39;DVL1&#39;: (0.09163004800347647, 0.35255399643853036),
 &#39;MAGEL2&#39;: (0.5926688876223712, 2.729502948534346e-11),
 &#39;SLC6A4&#39;: (-0.008573793435634278, 0.9308263894863134),
 &#39;SGSH&#39;: (0.1400033418271194, 0.15431322372547143),
 &#39;SHANK3&#39;: (0.09513931972764542, 0.33433944229094237)}
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">dpi</span><span class="o">=</span><span class="mi">400</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">labelsize</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>

<span class="k">for</span> <span class="n">spine</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">spine</span><span class="o">.</span><span class="n">set_position</span><span class="p">((</span><span class="s1">&#39;outward&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">4.5</span><span class="p">,</span><span class="mf">4.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">])</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mf">3.5</span><span class="p">,</span> <span class="mf">3.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="o">-</span><span class="mi">2</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">])</span>

<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">25</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">regplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">stats</span><span class="o">.</span><span class="n">zscore</span><span class="p">(</span><span class="n">mutation_trans_in_human</span><span class="p">[</span><span class="s1">&#39;Magel2&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">()),</span><span class="n">y</span><span class="o">=</span><span class="n">stats</span><span class="o">.</span><span class="n">zscore</span><span class="p">(</span><span class="n">AHBA_ex</span><span class="o">.</span><span class="n">T</span><span class="p">[</span><span class="s1">&#39;MAGEL2&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">flatten</span><span class="p">()),</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="n">scatter_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;color&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;linewidths&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span><span class="s1">&#39;alpha&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="s1">&#39;s&#39;</span><span class="p">:</span><span class="mi">40</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_30_0.png" src="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_30_0.png" />
</div>
</div>
</section>
<section id="load-behavior-data">
<h2>load behavior data<a class="headerlink" href="#load-behavior-data" title="Permalink to this heading"></a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">autism_meta</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;meta_info_abide.csv&#39;</span><span class="p">)</span>
<span class="n">autism_meta</span><span class="p">[</span><span class="s1">&#39;SUB_ID&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;sub-00</span><span class="si">{}</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">autism_meta</span><span class="p">[</span><span class="s1">&#39;SUB_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">]</span>
<span class="n">autism_meta</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;SUB_ID&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">autism_info</span> <span class="o">=</span> <span class="n">human_deviation_pattern</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">values</span>
<span class="n">autism_meta</span> <span class="o">=</span> <span class="n">autism_meta</span><span class="o">.</span><span class="n">T</span><span class="p">[</span><span class="n">autism_info</span><span class="p">]</span><span class="o">.</span><span class="n">T</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">behavior_l</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;FIQ&#39;</span><span class="p">,</span><span class="s1">&#39;VIQ&#39;</span><span class="p">,</span><span class="s1">&#39;PIQ&#39;</span><span class="p">,</span><span class="s1">&#39;ADOS_COMM&#39;</span><span class="p">,</span><span class="s1">&#39;ADOS_SOCIAL&#39;</span><span class="p">,</span><span class="s1">&#39;ADOS_STEREO_BEHAV&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">autism_meta</span> <span class="o">=</span> <span class="n">autism_meta</span><span class="p">[</span><span class="n">behavior_l</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">autism_meta</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>FIQ</th>
      <th>VIQ</th>
      <th>PIQ</th>
      <th>ADOS_COMM</th>
      <th>ADOS_SOCIAL</th>
      <th>ADOS_STEREO_BEHAV</th>
    </tr>
    <tr>
      <th>SUB_ID</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>sub-0050952</th>
      <td>134.0</td>
      <td>139.0</td>
      <td>120.0</td>
      <td>7.0</td>
      <td>10.0</td>
      <td>4.0</td>
    </tr>
    <tr>
      <th>sub-0050953</th>
      <td>132.0</td>
      <td>139.0</td>
      <td>117.0</td>
      <td>3.0</td>
      <td>10.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>sub-0050954</th>
      <td>78.0</td>
      <td>79.0</td>
      <td>80.0</td>
      <td>3.0</td>
      <td>8.0</td>
      <td>4.0</td>
    </tr>
    <tr>
      <th>sub-0050955</th>
      <td>93.0</td>
      <td>96.0</td>
      <td>91.0</td>
      <td>3.0</td>
      <td>11.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>sub-0050956</th>
      <td>115.0</td>
      <td>115.0</td>
      <td>110.0</td>
      <td>2.0</td>
      <td>8.0</td>
      <td>5.0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>sub-0050528</th>
      <td>100.0</td>
      <td>90.0</td>
      <td>109.0</td>
      <td>5.0</td>
      <td>7.0</td>
      <td>1.0</td>
    </tr>
    <tr>
      <th>sub-0050529</th>
      <td>128.0</td>
      <td>122.0</td>
      <td>127.0</td>
      <td>4.0</td>
      <td>9.0</td>
      <td>1.0</td>
    </tr>
    <tr>
      <th>sub-0050530</th>
      <td>106.0</td>
      <td>120.0</td>
      <td>91.0</td>
      <td>3.0</td>
      <td>10.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>sub-0050531</th>
      <td>112.0</td>
      <td>104.0</td>
      <td>119.0</td>
      <td>1.0</td>
      <td>11.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>sub-0050532</th>
      <td>84.0</td>
      <td>87.0</td>
      <td>81.0</td>
      <td>2.0</td>
      <td>6.0</td>
      <td>-9999.0</td>
    </tr>
  </tbody>
</table>
<p>233 rows × 6 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">magel2_scores_based_on_mouse</span> <span class="o">=</span> <span class="n">cross_species_autism_dataframe</span><span class="p">[</span><span class="s1">&#39;Magel2&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gene_behavior_l</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">gene_clinic_dataframe</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>

<span class="k">for</span> <span class="n">name_</span> <span class="ow">in</span> <span class="n">autism_meta</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">:</span>

    <span class="n">behavior_pattern</span> <span class="o">=</span> <span class="n">autism_meta</span><span class="p">[</span><span class="n">name_</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>

    <span class="n">behavior_pattern</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">isdigit</span><span class="p">()</span> <span class="k">else</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">behavior_pattern</span><span class="p">])</span>
    <span class="n">mask</span> <span class="o">=</span> <span class="o">~</span><span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">behavior_pattern</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">behavior_pattern</span><span class="o">!=-</span><span class="mf">9999.0</span><span class="p">)</span>

    <span class="n">arr1_filtered</span> <span class="o">=</span> <span class="n">magel2_scores_based_on_mouse</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span>
    <span class="n">arr2_filtered</span> <span class="o">=</span> <span class="n">behavior_pattern</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span>

    <span class="n">corr</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">arr1_filtered</span><span class="p">,</span><span class="n">arr2_filtered</span><span class="p">)</span>

    <span class="n">gene_behavior_l</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">corr</span><span class="p">)</span>

<span class="n">gene_clinic_dataframe</span><span class="p">[</span><span class="s1">&#39;magel2&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">gene_behavior_l</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gene_clinic_dataframe</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>magel2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>(-0.17657536932333698, 0.00701340405507845)</td>
    </tr>
    <tr>
      <th>1</th>
      <td>(-0.07795466860082743, 0.2358864146002949)</td>
    </tr>
    <tr>
      <th>2</th>
      <td>(-0.18717492106148584, 0.004224509163950006)</td>
    </tr>
    <tr>
      <th>3</th>
      <td>(-0.08570567055260474, 0.25533156937957996)</td>
    </tr>
    <tr>
      <th>4</th>
      <td>(0.0025750174236128893, 0.9727870157319674)</td>
    </tr>
    <tr>
      <th>5</th>
      <td>(-0.018771597619276183, 0.8046910213502325)</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">behavior_pattern</span> <span class="o">=</span> <span class="n">autism_meta</span><span class="p">[</span><span class="s1">&#39;FIQ&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<span class="n">behavior_pattern</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">isdigit</span><span class="p">()</span> <span class="k">else</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">behavior_pattern</span><span class="p">])</span>
<span class="n">mask</span> <span class="o">=</span> <span class="o">~</span><span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">behavior_pattern</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">behavior_pattern</span><span class="o">!=-</span><span class="mf">9999.0</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[33]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">arr1_filtered</span> <span class="o">=</span> <span class="n">magel2_scores_based_on_mouse</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span>
<span class="n">arr2_filtered</span> <span class="o">=</span> <span class="n">behavior_pattern</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[34]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">arr1_filtered</span><span class="p">,</span><span class="n">arr2_filtered</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[34]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PearsonRResult(statistic=-0.17657536932333698, pvalue=0.00701340405507845)
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[35]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">dpi</span><span class="o">=</span><span class="mi">400</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">labelsize</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>

<span class="k">for</span> <span class="n">spine</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">spine</span><span class="o">.</span><span class="n">set_position</span><span class="p">((</span><span class="s1">&#39;outward&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>

<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_linewidth</span><span class="p">(</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">25</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">regplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">arr2_filtered</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">arr1_filtered</span><span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="n">scatter_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;color&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;#669bbc&#39;</span><span class="p">,</span> <span class="s1">&#39;linewidths&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span><span class="s1">&#39;alpha&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="s1">&#39;s&#39;</span><span class="p">:</span><span class="mi">40</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_42_0.png" src="../../../../_images/tutorials_notebooks_translation_autism_mutation_gene_mutations_42_0.png" />
</div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../optogenetic_annotation/optogenetic_circuits.html" class="btn btn-neutral float-left" title="Annotating optogenetic circuits" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../../../../api/utils.html" class="btn btn-neutral float-right" title="Utils" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, TransBrain Developers.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>